### PR TITLE
BREAKING CHANGE:  US135601: [Adoption] Lit 2 migration - Phase 2

### PR DESCRIPTION
**Pull request by Vitalii-Misechko**
[US135601](https://rally1.rallydev.com/#/?detail=/userstory/623256592977&fdp=true): [Adoption] Lit 2 migration - Phase 2

This PR makes the breaking changes required to update to Lit 2 (`lit-html` 2 and `lit-element` 3). The Lit 2 upgrade will be a coordinated effort, so **this PR should not merge at this time**.

Breaking change checklist: 

- [x] Update `lit-html` -> `^2`, `lit-element` -> `^3`, `open-wc/testing` -> `^3` (if used)
- [x] `createRenderRoot` usages -> move from LitElement to ReactiveElement 
- [x] Remove Dependabot Lit rules in `.github/dependabot.yml`
- [x] Bump any dependencies to their Lit 2 versions
- [x] Remove any manual lifecycle calls to Lit 1 reactive controllers (Eg. `this._subscriptionController.hostConnected()`, etc)
- [x] backwards-compatible changes merged

Renamed API's:
- [x] `UpdatingElement` -> `ReactiveElement`
- [x] `@internalProperty` -> `@state` (we don't use TypeScript decorators)
- [x] `static getStyles()` -> `static finalizeStyles(styles)`
- [x] `_getUpdateComplete()` -> `getUpdateComplete()`
- [x] `NodePart` -> `ChildPart`

Testing checklist: 
- [x] CI/ unit tests pass
- [x] Local testing on demo pages
- [x] Testing component in the LMS
   - [x] All dependencies point to Lit 2 version
   - [x] The dashboard loads without errors
   - [x] Export to CSV and Learn More work correctly
   - [x] ! OrgUnits tree doesn't have any data. This is true for the main branch too 

For more information on the Lit 2 upgrade, visit https://lit.dev/docs/releases/upgrade/ or contact team Gaudi.

### CODE REVIEW

**Comment by Vitalii-Misechko** [comment]
The latest commit breaks CI. **It is expected**

**Review by annabelsegalld2l — APPROVED** [review]

**Review by johngwilkinson — DISMISSED** [review]

**Comment by Vitalii-Misechko** [comment]
please re-approve

File: package.json [hunk 1]
```diff
@@ -53,14 +53,14 @@
...
   "dependencies": {
-    "@adobe/lit-mobx": "0.0.x",
-    "@brightspace-ui-labs/card-overlay": "^1.0.0",
-    "@brightspace-ui-labs/immersive-nav": "^1.0.3",
-    "@brightspace-ui-labs/lit-router": "^1.0.9",
-    "@brightspace-ui-labs/ou-filter": "^1",
-    "@brightspace-ui-labs/pagination": "^1",
-    "@brightspace-ui-labs/toaster": "^1.0.2",
-    "@brightspace-ui/core": "^1.179.1",
+    "@adobe/lit-mobx": "^1.0.0",
```
Comment (annabelsegalld2l) [inline package.json:56]: This should be ^2.

Comment (Vitalii-Misechko) [reply package.json:56]: According [the link ](https://github.com/adobe/lit-mobx/pull/93/files?diff=split&w=1)you have provided to me it should be `1.0.0`, right?

```md
This library has been updated to the latest version of Lit which we highly recommend you update to. 
However if you wish to continue using LitMobx with [LitElement 2.0](https://lit-element.polymer-project.org/guide)
then install the 1.0 major version:
```

Comment (annabelsegalld2l) [reply package.json:56]: We are moving to Lit-element 3, so we need version 2. LitElement 3.0 is Lit 2.0. We have been calling the migration Lit 2, which moves LitElement to 3 and LitHtml to 2. I noticed the other dashboards had this same issue. We can just merge a one-line PR to bump those afterwards.

Comment (Vitalii-Misechko) [reply package.json]: Oh...thanks. Fixed

**Review by annabelsegalld2l — APPROVED** [review]

**Review by johngwilkinson — APPROVED** [review]

File: package.json [hunk 2]
```diff
@@ -53,14 +53,14 @@
...
   "dependencies": {
-    "@adobe/lit-mobx": "0.0.x",
-    "@brightspace-ui-labs/card-overlay": "^1.0.0",
-    "@brightspace-ui-labs/immersive-nav": "^1.0.3",
-    "@brightspace-ui-labs/lit-router": "^1.0.9",
-    "@brightspace-ui-labs/ou-filter": "^1",
-    "@brightspace-ui-labs/pagination": "^1",
-    "@brightspace-ui-labs/toaster": "^1.0.2",
-    "@brightspace-ui/core": "^1.179.1",
+    "@adobe/lit-mobx": "^2.0.0",
```
[the remark this answers is not in the document]
Comment (annabelsegalld2l) [reply package.json:56]: This update is needed in the 3 dashboards and ou-filter, which Vitalii is handling. I'm working on activities and content-components, and that should be all the usages 😄

**Pull request merged**